packages/apps/plugins/plugin-graph/src/graph/graph-builder.ts

```diff
@@ -0,0 +1,178 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import { RevertDeepSignal, deepSignal } from 'deepsignal/react';
+
+import { SendIntent } from '@braneframe/plugin-intent';
+import { EventSubscriptions } from '@dxos/async';
+
+import { GraphStore } from './graph';
+import { Graph } from './types';
+
+/**
+ * The builder...
+ */
+export class GraphBuilder {
+  private readonly _nodeBuilders = new Set<Graph.NodeBuilder>();
+  private readonly _unsubscribe = new Map<string, EventSubscriptions>();
+
+  private _sendIntent?: SendIntent;
+
+  // TODO(burdon): add/remove or register/unregister.
+  registerNodeBuilder(builder: Graph.NodeBuilder): void {
+    this._nodeBuilders.add(builder);
+  }
+
+  removeNodeBuilder(builder: Graph.NodeBuilder): void {
+    this._nodeBuilders.delete(builder);
+  }
+
+  build(root?: Graph.Node, path: string[] = [], ignoreBuilders: Graph.NodeBuilder[] = []): GraphStore {
+    const graph = new GraphStore(root ?? this._createNode({ id: 'root', label: 'Root' }), path);
```

Comment:
This function is being called recursively whenever a node is added and so you're making a new graph store for every single node. I'm assuming that's not intended?